clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/HelixParticipationService.java

```diff
@@ -310,11 +314,24 @@ private void asyncStart() {
         veniceServerConfig.getPubSubClientsFactory().getProducerAdapterFactory();
     VeniceWriterFactory writerFactory =
         new VeniceWriterFactory(veniceProperties.toProperties(), pubSubProducerAdapterFactory, null);
-    String dummyPushStatusStoreName = VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.getSystemStoreName("dummy");
-    SchemaEntry valueSchemaEntry =
-        helixReadOnlySchemaRepository.getSupersetOrLatestValueSchema(dummyPushStatusStoreName);
-    DerivedSchemaEntry updateSchemaEntry =
-        helixReadOnlySchemaRepository.getLatestDerivedSchema(dummyPushStatusStoreName, valueSchemaEntry.getId());
+    SchemaEntry valueSchemaEntry;
+    DerivedSchemaEntry updateSchemaEntry;
+    try {
+      String dummyPushStatusStoreName = VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.getSystemStoreName("dummy");
+      valueSchemaEntry = helixReadOnlySchemaRepository.getSupersetOrLatestValueSchema(dummyPushStatusStoreName);
+      updateSchemaEntry =
+          helixReadOnlySchemaRepository.getLatestDerivedSchema(dummyPushStatusStoreName, valueSchemaEntry.getId());
+    } catch (VeniceNoStoreException e) {
+      LOGGER.warn(
+          "ZK-shared system store is not available in current environment, will fall back to last known protocol version instead.");
+      int valueSchemaId = AvroProtocolDefinition.PUSH_STATUS_SYSTEM_SCHEMA_STORE.getCurrentProtocolVersion();
+      Schema valueSchema = AvroProtocolDefinition.PUSH_STATUS_SYSTEM_SCHEMA_STORE.getCurrentProtocolVersionSchema();
+      valueSchemaEntry = new SchemaEntry(valueSchemaId, valueSchema);
+      updateSchemaEntry = new DerivedSchemaEntry(
+          valueSchemaId,
+          1,
```

Comment:
Is this the WC protocol version? Is it safe to hardcode this?

Reply:
We were using this: veniceProperties.getInt(PUSH_STATUS_STORE_DERIVED_SCHEMA_ID, 1) actually and we never set the value so it was 1 all the time.

I believe we should never hit the fall back case when we really need to use push status store. Because the ZK shared store is not in the cluster, that means any r/w to push status store should fail, as user push status store should not be there.

Reply:
Maybe we should disable the subsequent write instead...